extensions/notebook/src/book/bookPinManager.ts

```diff
@@ -33,10 +33,7 @@ export class BookPinManager implements IBookPinManager {
 	}
 
 	isNotebookPinned(notebookPath: string): boolean {
-		if (getPinnedNotebooks().findIndex(x => x === notebookPath) > -1) {
-			return true;
-		}
-		return false;
+		return (getPinnedNotebooks().filter(function (x) { return x.notebookPath === notebookPath; }).length > 0);
```

Comment:
Any particular reason you made this into a function object? Lamba's are generally easier to read. Any why did you change the logic anyways? The old check seemed fine if you just updated it to be x.notebookPath

Reply:
I'm not sure why I did this change 😅 . But yeah I agree, I'll change it back.